common/paramsbuilder/params.go

```diff
@@ -19,9 +19,19 @@ type ParamAssurance interface {
 var (
 	ErrMissingClient     = errors.New("http client not set")
 	ErrMissingWorkspace  = errors.New("missing workspace name")
+	ErrMissingRegion     = errors.New("missing region name")
 	ErrNoSupportedModule = errors.New("no supported module was chosen")
 )
 
+// Create will apply options to construct a ready to go set of parameters.
+func Create[P ParamAssurance](params P, opts []func(params *P)) (*P, error) {
```

Comment:
I came up with a generic constructor for connector parameters. Later we could replace with this statement and make things shorter.

Reply:
If this is not being used in the current PR, let's introduce it later as it is used

Reply:
I have pulled this change into a dedicated small PR: https://github.com/amp-labs/connectors/pull/711
This change will be removed from this PR for now.

Reply:
Thanks! Please resolve this thread when done.

Reply:
This PR is stacked so it will use this constructor but not introduce.